swagger.yaml

```diff
@@ -0,0 +1,226 @@
+swagger: "2.0"
+info:
+  title: "Tweet API"
+  description: "This provides an API for creating a message (post), reading a message (get), editing a message (put), delete a message (delete)."
+  version: "1.0"
+  license:
+    name: GNU Affero General Public License Version 3
+    url: http://www.gnu.org/licenses/agpl-3.0.en.html
+externalDocs:
+  description: GitHub Repository
+  url: https://github.com/osu-mist/tweet-api
+host: localhost
+schemes:
+  - "https"
+basePath: /api/v1
+consumes:
+  - application/json
+produces:
+  - application/json
+
+tags:
+- name: "tweets"
+
+paths:
+
+  /tweets:
+
+    get:
+      tags:
+      - tweets
+      description: "search for a user's tweets"
+      parameters:
+      - in: "query"
+        name: userName
+        required: false
+        type: string
+
+      - in: "query"
+        name: id
+        required: false
+        type: string
+
+      responses:
+        200:
+          description: Successful response
+          schema:
+            $ref: "#/definitions/userTweets"
+        400:
+          description: The specified user ID is invalid
+          schema:
+            $ref: "#/definitions/Error"
```

Comment:
When returning error/errors, by the definition in [JSON API](http://jsonapi.org/format/#errors):

> Error objects MUST be returned as an array keyed by errors in the top level of a JSON API document.

Example: https://github.com/osu-mist/express-api-skeleton/blob/feature/CO-1192-express-api-skeleton/swagger.yaml#L73